Airgap howto. CNCF suggested language improvements. (#651)
kubewarden/docs: docs/howtos/airgap/02-install.md

keywords: [kubewarden, kubernetes, air gap installation]
doc-persona: [kubewarden-operator, kubewarden-integrator]
doc-type: [howto]
doc-topic: [operator-manual, air gap, installation]
---

<head>
  <link rel="canonical" href="https://docs.kubewarden.io/howtos/airgap/install"/>
</head>

This guide shows you how to install Kubewarden in air-gapped environments. For
an air-gapped installation of Kubewarden, you need a private Open Container
Initiative (OCI) registry accessible by your Kubernetes cluster. Kubewarden

> Policies are WebAssembly modules, therefore you can store them in an

Reviewer: ```suggestion
Policies are WebAssembly modules; therefore, you can store them in an
```

      --images kubewarden-images.tar.gz
    ```

    Docker begins pulling the images used for an air gap install. Be patient.
    This process takes a few minutes. When complete, your current directory,
    where you ran the command, has a tarball, `kubewarden-images.tar.gz`.

## Save policies in your workstation

> 1. Add all the policies you want to use in a `policies.txt` file. A file with a
>    list of default policies is in the Kubewarden defaults [release
>    page](https://github.com/kubewarden/helm-charts/releases/).

Reviewer: ```suggestion
1. Add all the policies you want to use to a `policies.txt` file. A file with a
   list of default policies is in the Kubewarden defaults [release
   page](https://github.com/kubewarden/helm-charts/releases/).
```

- `kubewarden-load-policies.sh` and
- `policies.txt`

1. Load Kubewarden images into the private registry. You need to authenticate
   the Docker client against the local registry.

   ```shell
   ./kubewarden-load-images.sh \
      --image-list ./kubewarden-images.txt \
      --images kubewarden-images.tar.gz \
      --registry <REGISTRY.YOURDOMAIN.COM:PORT>
   ```

1. Load Kubewarden policies into the private registry. You should authenticate
   `kwctl` against the local registry (`kwctl` uses the same mechanism to
   authenticate as `docker`, a `~/.docker/config.json` file)

   ```shell
   ./kubewarden-load-policies.sh \
      --policies-list policies.txt \
      --policies kubewarden-policies.tar.gz \
      --registry <REGISTRY.YOURDOMAIN.COM:PORT> \
      --sources-path sources.yml
   ```

:::caution

The `kwctl` command needs the `sources.yaml` file to connect to registries in
these categories:

- Authentication is required.

> - Self signed certificate is being used.

Reviewer: ```suggestion
- Self-signed certificate is being used.
```

- No TLS termination is done.

Please refer to the [section on custom certificate
authorities](../custom-certificate-authorities.md) in the documentation to

> learn how to configure the `sources.yaml` file

Reviewer: ```suggestion
learn how to configure the `sources.yaml` file.
```

```

:::caution

> To use the Policy Reported sub-chart available in the `kubewarden-controller`
> chart you need to define other values specific for the sub-chart in an
> air-gapped environment. See an example below:

Reviewer: ```suggestion
To use the Policy Reported subchart available in the `kubewarden-controller`
chart, you need to define other values specific for the subchart in an
air-gapped environment. See an example below:
```

    --set policy-reporter.ui.image.repository=kyverno/policy-reporter-ui
```

> It's necessary to define `auditScanner.policyReporter` to enable the sub-chart
> and 4 more values, to configure the registry and repository where you store the
> Policy Reporter images. For more information about the policy report sub-chart
> values see [Policy Reporter
> documentation](https://kyverno.github.io/policy-reporter-docs/getting-started/helm.html).

Reviewer: ```suggestion
It's necessary to define `auditScanner.policyReporter` to enable the subchart
and 4 more values to configure the registry and repository where you store the
Policy Reporter images. For more information about the policy report subchart
values, see [Policy Reporter
documentation](https://kyverno.github.io/policy-reporter-docs/getting-started/helm.html).
```

```

:::caution

To download the recommended policies installed by the `kubewarden-defaults`
Helm Chart from a registry other than `global.cattle.systemDefaultRegistry`,
you can use the `recommendedPolicies.defaultPoliciesRegistry` configuration.
This configuration lets users specify a registry dedicated to pulling the OCI
artifacts of the policies. It's particularly useful when their container image
repository doesn't support OCI artifacts.

> To install, and wait for the installation to complete, use the following

Reviewer: ```suggestion
To install and wait for the installation to complete, use the following
```
